src/bin/inx-chronicle/stardust_inx/mod.rs

```diff
@@ -373,11 +384,39 @@ async fn handle_cone_stream(db: &MongoDb, inx: &mut Inx, milestone_index: Milest
         .try_collect::<Vec<_>>()
         .await?;
 
+    let stats = blocks_with_metadata.iter().fold(
```

Comment:
Correct me if I am wrong, but isn't `inx.read_milestone_cone(..)` exactly providing us with those blocks?

Reply:
On the current testnet this endpoint returns:
```
{"blocks":59,"perPayloadType":{"noPayload":0,"txsConfirmed":2,"txsConflicting":14,"taggedData":42,"milestone":1}}
```
for milestone id `0xbf6bc9124c602d60ca55e8323c39b505c762d2d7c221783870b12880cba4c0e4`

Reply:
Depends. If the past cone is limited only to that milestone, then I suppose that is correct. But if it is supposed to include previous milestones in the cone as well, then it is not sufficient.

Reply:
Yea. But as far as I know the definition of a `milestone past-cone` is still basically: 
All vertices that are directly or indirectly referenced by a milestone X that are not already referenced by any milestone < X.

Reply:
Perhaps we should clarify this. If this is the case, then I don't see why we would need to pre-calculate this.

Reply:
`inx.read_milestone.cone()` only returns the blocks within the milestone's past cone up to the point where it would hit blocks which were referenced by a previous milestone.

I guess pre-computing makes sense here instead of having to fold/reduce blocks every time the endpoint is queried given that it could become more expensive if the cone is made up of thousands of blocks.

Reply:
This confirms my belief then. It's not enough to use the data from the milestone cone. We need to get the data from the milestone with `previous_milestone_id` then add on top of that.